src/platform/pal_error.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "pal_internal.h"
+#include "atomic_ref_count.h"
+
+namespace xlang
+{
+    namespace impl
+    {
+        struct error_info : xlang_error_info
+        {
+            explicit error_info(xlang_result result) noexcept
+                : m_result(result)
+            {}
+
+            int32_t XLANG_CALL QueryInterface(xlang_guid const& id, void** object) noexcept override
```

Comment:
These should all probably be `final`.

Reply:
Will do, but these are merely placeholder implementations for now. :)

Reply:
Ah right.